### PR TITLE
added tmux navigator

### PR DESCRIPTION
**Pull request by fabianpage**

### CODE REVIEW

File: plugins/utils/tmux-navigator.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  options.plugins.tmux-navigator = {
+    enable = mkEnableOption "Enable Tmux-Navigator (see https://github.com/christoomey/vim-tmux-navigator for tmux installation instruction)";
+
+    package = helpers.mkPackageOption "tmux-navigator" pkgs.vimPlugins.tmux-navigator;
+
+    tmuxNavigatorSaveOnSwitch = helpers.mkNullOrOption lib.types.int ''
+      null: don't save on switch
+      1: update (write the current buffer, but only if changed)
+      2: wall (write all buffers)
+    '';
```
Comment (traxys) [inline plugins/utils/tmux-navigator.nix:20]: If this is an enum couldn't you use a `helpers.mkNullOrOption (lib.types.enum [1 2])` to be more type aware ? Morever can you add to the description the upstream default value

File: plugins/utils/tmux-navigator.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    tmuxNavigatorDisableWhenZoomed = helpers.mkNullOrOption lib.types.int ''
```
Comment (traxys) [inline plugins/utils/tmux-navigator.nix:22]: Same as the other option

**Review by traxys — CHANGES_REQUESTED** [review]
Could you also add a test with the empty  configuration and the default configuration please ?

File: plugins/utils/tmux-navigator.nix (hunk 1 above)
Comment (fabianpage) [reply plugins/utils/tmux-navigator.nix:20]: ah nice, didn't knew this type.

File: plugins/utils/tmux-navigator.nix (hunk 2 above)
Comment (fabianpage) [reply plugins/utils/tmux-navigator.nix:22]: here only null or 1 is allowed, should i put helpers.mkNullOrOption (lib.types.enum [1]) ?

Comment (traxys) [reply plugins/utils/tmux-navigator.nix:22]: It's a bit verbose but yes I think it's usefull
